source/4.0/learn/introduction/index.html.md

```diff
@@ -1,24 +1,148 @@
 ---
 chapter: Introduction
 sections:
-  - philosophy
-  - why
   - active-record
 ---
 
-Ruby Object Mapper (ROM) is a Ruby persistence library with the goal to provide
-powerful object mapping capabilities without limiting the full power of your
-datastore.
+$TOC
+  1. [What is ROM](#what-is-rom)
+  2. [The Problem with ORMs](#the-problem-with-orms)
+  3. [Why use ROM](#why-use-rom)
+  4. [Principles & Design](#principles-amp-design)
+  5. [Our Inspiration](#inspirations-ideas-and-friends)
+$TOC
+
+## What is ROM
+
+Ruby Object Mapper (ROM) is a *fast* ruby persistence library with
+the goal of providing powerful object mapping capabilities without limiting the
+full *power* of the underlying datastore.
 
 More specifically, ROM exists to:
 
 * Isolate the application from persistence details
 * Provide minimum infrastructure for mapping and persistence
 * Provide shared abstractions for lower-level components
-* Provide simple use of the underlying datastore when desired
+* Provide simple use of *power* features offered by the datastore
+
+
+## The Problem with ORMs
+
+Object hierarchies are very different from relational hierarchies. Relational
+hierarchies focus on data and its relationships whereas objects not only manage
+data, but also their *identity* and the *behavior* centered around that data.
+When attempting to reconcile those difference and map one to the other, we as
+developers end up *splatting* against a concrete wall.
+
+We believe the cause can be attributed to a fundamental flaw behind ORMs; that
+is this idea that it's easy to:
+
+1. Map objects to database tables one-to-one (the
+[ActiveRecord](https://en.wikipedia.org/wiki/Active_record_pattern) design
+pattern); or
+
+2. Introduce machinery to translate between objects and persistence structures
+(the [DataMapper](https://en.wikipedia.org/wiki/Data_mapper_pattern) pattern)
+
+Both strategies tend to start out fine, but can quickly become cumbersome as an
+application transitions to a medium-to-large application.
+
+ActiveRecord style mapping has the benefit of aiding in rapid prototyping works
+great in basic CRUD scenarios but it limits application modeling to what's
+convenient for the database. Entities will tend to map one-to-one with tables
+which leads to knowledge of the persistence structure infecting the application
+domain leading to change resistant software.
```

Comment:
What does `leading to change resistant software.` mean in this context? As a non-native speaker, I found it hard to understand for some reason 😓